src/structures/GuildPreview.js

```diff
@@ -0,0 +1,154 @@
+'use strict';
+
+const Base = require('./Base');
+const GuildEmojiManager = require('../managers/GuildEmojiManager');
+
+/**
+ * Represents the data about the guild any bot can preivew, connected to the specified public guild
+ */
+class GuildPreview extends Base {
+  constructor(client, data) {
+    super(client);
+
+    if (!data) return;
+    this._patch(data);
+  }
+
+  /**
+   * Builds the public guild with the provided data
+   * @param {*} data The raw data of the public guild
+   * @private
+   */
+  _patch(data) {
+    /**
+     * The id of this public guild
+     * @type {string}
+     */
+    this.id = data.id;
+
+    /**
+     * The name of this public guild
+     * @type {string}
+     */
+    this.name = data.name;
+
+    /**
+     * The icon of this public guild
+     * @type {?string}
+     */
+    this.icon = data.icon;
+
+    /**
+     * The splash icon of this public guild
+     * @type {?string}
+     */
+    this.splash = data.splash;
+
+    /**
+     * The discovery splash icon of this public guild
+     * @type {?string}
+     */
+    this.discoverySplash = data.discovery_splash;
+
+    /**
+     * An array of enabled guild features
+     * @type {Features[]}
+     */
+    this.features = data.features;
+
+    /**
+     * The approximate count of members in this public guild
+     * @type {number}
+     */
+    this.approximateMemberCount = data.approximate_member_count;
+
+    /**
+     * The approximate count of online members in this public guild
+     * @type {number}
+     */
+    this.approximatePresenceCount = data.approximate_presence_count;
+
+    /**
+     * The description for this public guild
+     * @type {?string}
+     */
+    this.description = data.description;
+
+    if (!this.emojis) {
+      /**
+       * A manager of the emojis belonging to this public guild
+       * @type {GuildEmojiManager}
+       */
+      this.emojis = new GuildEmojiManager(this);
+      if (data.emojis) for (const emoji of data.emojis) this.emojis.add(emoji);
+    } else if (data.emojis) {
+      this.client.actions.GuildemojisUpdate.handle({
+        guild_id: this.id,
+        emojis: data.emojis,
+      });
```

Comment:
If the guild is not handled by the client this would be a noop.
If it is handled this would potentially modify the emoji cache of the `Guild` in `Client#guilds` and not this `GuildPreview`!
See:
https://github.com/discordjs/discord.js/blob/df324e2c21171aa17bc4e43f4a36f78c2f0eaec1/src/client/actions/GuildEmojisUpdate.js#L7-L8

A solution to this could be done as the following:
- if `this.emojis` does not exist, create it.
- else, [`clear`](https://discord.js.org/#/docs/collection/master/class/Collection?scrollTo=clear) its [`cache`](https://discord.js.org/#/docs/main/stable/class/GuildEmojiManager?scrollTo=cache).
  It does not matter what emojis were added or removed here after all.
- And then, if emojis exist in data, (re-)[`add`](https://github.com/discordjs/discord.js/blob/stable/src/managers/GuildEmojiManager.js#L30) all emojis.

Reply:
`this.emojis` will never exist as this structure is never cached. I think just a simple `this.emojis = new ...` followed by adding the emojis will suffice

Reply:
Maybe we shouldn't add a `GuildEmojiManager` to this structure, as there's no guarantee that the client can interact with the guild (create new emojis, for example), since it's a guild available to everyone via the `preview` endpoint. Perhaps build a separate manager structure specifically for this guild can be done.

Reply:
Although when I think about it, this kind of guild is never going to exist in the client's cache (that is in a scenario where the client isn't a part of said guild), so there's not much point in creating a manager for it, or assigning one at all. Perhaps there's just a bit of utility we'd need from the managers. But I don't think we'd necessarily have to create a cache for this structure.

Reply:
Another thing, it'd probably be necessary to create an interface to describe the structure of a single emoji that can be found in that array received from the payload. Then assign it as an array of that interface.
I've created an interface that corresponds to the structure of a single object in the `emojis` array.
```ts
  interface GuildPreviewEmojis {
    name: string;
    roles: Role[];
    id: string;
    require_colons: boolean;
    managed: boolean;
    animated: boolean;
    available: boolean;
  };
```
Let me know if I should go about this like so.

Reply:
The options that I currently see here are:
- **Raw data**
  Misses out on any kind of utility discord.js usually provides for emoji:
- - `identifier`
- - `url`
- - `createdAt`
- - `createdTimestamp`
- - `toString`
- **Regular [`Emoji`](https://github.com/discordjs/discord.js/blob/master/src/structures/Emoji.js#L10)**
  Would discard a few properties that are actually there:
- - `roles`
- - `requireColons`
- - `managed`
- - `available`
- **`GuildEmojiManager`**
  This one and its `GuildEmoji` would provide api methods that do not make too much sense to be there, but should still work, given the bot has relevant permissions.
BUT `GuildEmoji#roles` would throw since `GuildPreview#roles` does not exist.
- **A new  `GuildPreviewEmoji` class**
  To avoid redundancies it would be possible to create a common parent class for `GuildEmoji` and this class.
  This common parent would provide everything that `GuildEmoji` does, except api methods and the `roles` getter.
  `GuildPreviewEmoji` would then provide a `Set<Snowflake>` for the latter.

Reply:
I personally think that we should go for the `GuildPreviewEmoji` class, since it makes the most sense to do, and doesn't miss out on the utility that the library is suppose to suffice, aswell as not providing the raw data of the payload. So, the very last option seems like the way to go about this, sweet suggestion, will start working on it right now. :+1:

Reply:
Just one thing, 

When creating the `GuildPreviewEmoji` class, should I assign the properties? Since they're technically going to be there once I run `BaseEmoji#_patch(data)`. Or is the purpose of doing that solely for documentation?

Reply:
I'll be pushing the new changes, please do let me know if anything needs changing, or if the `GuildPreviewEmoji` object needs further JSDocs documentation.